File: users-service/src/main/java/com/psiboard/users_service/application/dto/UserRequestDto.java
package com.psiboard.users_service.application.dto;

import com.psiboard.users_service.domain.UserRole;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserRequestDto {
    @NotBlank(message = "Nome não pode ser vazio!")
    @NotNull(message = "Nome não pode ser nulo!")
    private String name;

    @NotBlank(message = "Email não pode ser vazio!")
    @Email(message = "Digite um email válido")
    private String email;

    @Size(min = 6, message = "A senha precisa ter no minimo 6 caracteres!")
    private String password;

    @NotBlank(message = "Contato não pode ser vazio!")
    @NotNull(message = "Contato não pode ser nulo!")
    private String contact;
    
    // @NotNull(message = "Role não pode ser nulo!")
    // @NotBlank(message = "Role não pode ser vazio!")
    private UserRole role;
}

File: client-bff/src/app/commons/dto/request/user-request.dto.ts
import { IsEmail, IsNotEmpty, IsString, MinLength, IsEnum } from 'class-validator'; 
import { UserRole } from '../../enums/user-role.enum';

export class UserRequestDto {
  
  @IsNotEmpty({ message: 'Nome não pode ser vazio!' })
  @IsString({ message: 'Nome deve ser uma string!' })
  name: string;

  @IsNotEmpty({ message: 'Email não pode ser vazio!' })
  @IsEmail({}, { message: 'Digite um email válido!' })
  email: string;

  @MinLength(6, { message: 'A senha precisa ter no mínimo 6 caracteres!' })
  password: string;

  @IsNotEmpty({ message: 'Contato não pode ser vazio!' })
  @IsString({ message: 'Contato deve ser uma string!' })
  contact: string;

  @IsEnum(UserRole, { message: 'Role inválido!' })
  role: UserRole;
}
